e2e/internal/keyexec/keyexec.go

```diff
@@ -0,0 +1,361 @@
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package keyexec
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	expect "github.com/Netflix/go-expect"
+	"github.com/sylabs/singularity/e2e/internal/e2e"
+	"github.com/sylabs/singularity/internal/pkg/test"
+	"github.com/sylabs/singularity/pkg/sypgp"
+)
+
+type testingEnv struct {
+	// base env for running tests
+	CmdPath     string `split_words:"true"`
+	TestDir     string `split_words:"true"`
+	RunDisabled bool   `default:"false"`
+}
+
+var testenv testingEnv
+
+// PullDefaultPublicKey will pull the public Sylabs Admin key
+func PullDefaultPublicKey(t *testing.T) {
+	e2e.LoadEnv(t, &testenv)
+
+	argv := []string{"key", "pull", "F69C21F759C8EA06FD32CCF4536523CE1E109AF3"}
+
+	execKey := exec.Command(testenv.CmdPath, argv...)
+
+	out, err := execKey.CombinedOutput()
+	if err != nil {
+		t.Log(string(out))
+		t.Fatalf("Unable to pull key: %v", err)
+	}
+}
+
+// RemoveDefaultPublicKey will pull the public Sylabs Admin key
+func RemoveDefaultPublicKey(t *testing.T) {
+	e2e.LoadEnv(t, &testenv)
+
+	argv := []string{"key", "remove", "F69C21F759C8EA06FD32CCF4536523CE1E109AF3"}
+	execKey := exec.Command(testenv.CmdPath, argv...)
+
+	out, err := execKey.CombinedOutput()
+	if err != nil {
+		t.Log(string(out))
+		t.Fatalf("Unable to pull key: %v", err)
+	}
+}
+
+func RemovePublicKeyring(t *testing.T) {
+	err := os.Remove(sypgp.PublicPath())
+	if err != nil {
+		t.Fatalf("Unable to remove public keyring: %v", err)
+	}
+}
+
+// RemoveSecretKeyring will delete your secret keyring.
+func RemoveSecretKeyring(t *testing.T) {
+	err := os.Remove(sypgp.SecretPath())
+	if err != nil {
+		t.Fatalf("Unable to remove secret keyring: %v", err)
+	}
+}
+
+func RemoveKeyring(t *testing.T) {
+	err := os.RemoveAll(sypgp.DirPath())
+	if err != nil {
+		t.Fatalf("Unable to remove keyring directory: %v", err)
+	}
+}
+
+// ImportKey will import a key from kpath.
+func ImportKey(t *testing.T, kpath string) ([]byte, error) {
+	e2e.LoadEnv(t, &testenv)
+
+	argv := []string{"key", "import", kpath}
+	execKey := exec.Command(testenv.CmdPath, argv...)
+
+	return execKey.CombinedOutput()
+}
+
+// ImportPrivateKey will take a private key file (kpath) and import it.
+func ImportPrivateKey(t *testing.T, kpath string) (string, []byte, error) {
+	e2e.LoadEnv(t, &testenv)
+
+	c, err := expect.NewConsole(expect.WithStdin(os.Stdin))
+	if err != nil {
+		panic(err)
+	}
+	defer c.Close()
+
+	exportCmd := []string{"key", "import", kpath}
+
+	cmd := exec.Command(testenv.CmdPath, exportCmd...)
+	cmd.Stdin = c.Tty()
+
+	buf := bytes.NewBuffer(nil)
+	cmd.Stderr = buf
+	cmd.Stdout = buf
+
+	go func() {
+		c.ExpectEOF()
+	}()
+
+	err = cmd.Start()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Send the passcode to singularity
+	c.Send("e2etests\n")
+	c.Send("e2etests\n")
+	c.Send("e2etests\n")
```

Comment:
Because when you import a private key, it will ask you for the old password, then a new password, and then to retype you password. For this test, its all using the same password.

Reply:
cool, could you add a comment to that? so that when somebody needs to maintain it in the future it could be clear